### DISCUSSION TITLE
What's the best way to have dynamic colours based on db values?

### DISCUSSION THREAD

**Discussion by n8udd** [tailwindlabs/tailwindcss#5799] [Help] [answered]
I'm trying to set the colour values of some buttons based on values in the database that are being passed to my React component as props (using Inertiajs), and create the relevant className accordingly. I know it wouldn't work with inline backticks, so I tried it as a function and that didn't work either.

The code works with `npm run dev` *without* JIT on, but fails with `nom run prod` both _with and without_ JIT mode.

The following is a gist with my component: https://gist.github.com/n8udd/6841daa7be007fe86216cb4ade7e095c

If I've used the classNames in other areas of the site it works, so I believe it's a compile error, but having added the colours to my safelist in the config, they still appear to be being purged.

Gist for tailwind config: https://gist.github.com/n8udd/27ed28477017259b3d9aa0137590bcde

**Comment by n8udd** [comment, marked as answer]
Turns out I was on v2.1 and needed to be on v2.2 as per [here](https://github.com/tailwindlabs/tailwindcss/issues/2577#issuecomment-874620351) for the safelist to work.

**Comment by jk-baseer** [comment]
Hey @n8udd 
Can i ask how did you achieve this? Have been searching about applying dynamic values to tailwind config from Server based on DB value or from a JSON file.
